cpp/src/ice2slice/Gen.cpp

```diff
@@ -299,28 +299,114 @@ namespace
         if (operation)
         {
             std::map<std::string, StringList> parameterDocs = comment->parameters();
-            ParameterList parameters = operation->parameters();
+
+            // Input parameters
+            ParameterList parameters = operation->inParameters();
             for (ParameterList::const_iterator it = parameters.begin(); it != parameters.end(); it++)
             {
                 ParameterPtr param = *it;
-                if (!param->isOutParam())
+                std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
+                if (q != parameterDocs.end())
                 {
-                    std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
-                    if (q != parameterDocs.end())
+                    out << nl << "/// @param " << param->name() << ": ";
+                    for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
                     {
-                        out << nl << "/// @param " << param->name() << ": ";
-                        for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
+                        if (r != q->second.begin())
                         {
-                            if (r != q->second.begin())
-                            {
-                                out << nl;
-                            }
-                            out << (*r);
-                            r++;
+                            out << nl;
+                            out << "/// ";
                         }
+                        out << (*r);
+                        r++;
                     }
                 }
             }
+
+            // Output parameters
+            parameters = operation->outParameters();
+            for (ParameterList::const_iterator it = parameters.begin(); it != parameters.end(); it++)
+            {
+                ParameterPtr param = *it;
+                std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
+                if (q != parameterDocs.end())
+                {
+                    out << nl << "/// @returns " << param->name() << ": ";
+                    for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
+                    {
+                        if (r != q->second.begin())
+                        {
+                            out << nl;
+                            out << "/// ";
+                        }
+                        out << (*r);
+                        r++;
+                    }
+                }
+            }
+
+            // Return value
+            StringList returnDocs = comment->returns();
+            if (!returnDocs.empty())
+            {
+                out << nl << "/// @returns";
+                if (operation->returnsMultipleValues())
```

Comment:
I am not a fan of the 'returnsMultipleValues()' function name, since an Ice-Slice operation can't do that.

The absence of doc-comments in Parser.h is also unfortunate.